client/src/components/DirectoryAddUpdate.jsx

```diff
@@ -51,29 +47,53 @@ export default function DirectoryAddUpdate() {
     // Validation
     const [errors, setErrors] = React.useState([]);
 
+    // While loading coop data from ID
+    const [loadingCoopData, setLoadingCoopData] = React.useState(false);
 
-    // Code for EmailJS, not using as of 4-7-21
-    // const sendEmail = (e) => {
-    //     e.preventDefault();
+    // Gets id from URL
+    const { id } = useParams();
+
+    const fetchCoopForUpdate = async () => {
+        console.log(id);
+        setLoadingCoopData(true);
+
+        try {
+            const res = await fetch(REACT_APP_PROXY + `/coops/${id}/`);
+            if(!res.ok) {
+                throw Error("Cannot access requested entity.")
+            }
+            const coopResults = await res.json();
+            console.log(coopResults);
+
+            setCoopName(coopResults.name ?coopResults.name : "");
+            setStreet(coopResults.addresses[0].formatted ? coopResults.addresses[0].formatted : "");
+            setCity(coopResults.addresses[0].locality.name ? coopResults.addresses[0].locality.name : "");
+            setState(coopResults.addresses[0].locality.state.code ? coopResults.addresses[0].locality.state.code : "");
+            setZip(coopResults.addresses[0].locality.postal_code ? coopResults.addresses[0].locality.postal_code : "");
+            // setCounty();
+            setCountry(coopResults.addresses[0].locality.state.country.code ? coopResults.addresses[0].locality.state.country.code : "");
+            // setAddressPublic();
+            setWebsites(coopResults.web_site ? coopResults.web_site : "");
+            // setContactName();
+            // setContactNamePublic();
+            setContactEmail(coopResults.email ? coopResults.email : "");
+            // setContactEmailPublic();
+            setContactPhone(coopResults.phone ? coopResults.phone.phone : "");
+            // setContactPhonePublic();
+            // Need to figure out how entity types look in data if there are multiple
+            setEntityTypes([coopResults.types[0]] ? [coopResults.types[0].name] : []);
+            // setScope();
+            // setTags([]);
+            // setDescEng("");
+            // setDescOther("");
+            setReqReason("update");
+        } catch (error) {
+            console.log(error);
+            setErrors(`Error: ${error.message}`)
+        }
+        setLoadingCoopData(false);
```

Comment:
Let's put this in a "finally" block just in case something (in the future) goes wrong in the "catch" block.

Reply:
Done!